### PR TITLE
Update bcftools: v1.15.1

### PR DESCRIPTION
**Pull request by gallardoalba**
FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [ ] - This PR adds a new tool or tool collection
* [x] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

This PR updates BCFtools up to v1.15.1 and includes the PR https://github.com/galaxyproject/tools-iuc/pull/4434.

### CODE REVIEW

File: tools/bcftools/bcftools_plugin_setgt.xml [hunk 1]
```diff
@@ -52,11 +52,16 @@ bcftools plugin @EXECUTABLE@
                 <option value="./x">./x partially missing (./0 or .|1)</option>
                 <option value=".">. partially or completely missing</option>
                 <option value="a">a all genotypes</option>
+                <option value="b">b heterozygous genotypes failing two-tailed binomial test</option>
+                <option value="q">q select genotypes using -i/-e options</option>
+
```
Comment (nsoranzo) [inline tools/bcftools/bcftools_plugin_setgt.xml:57]:
```suggestion
```

File: tools/bcftools/bcftools_plugin_setgt.xml [hunk 2]
```diff
@@ -33,14 +33,14 @@ bcftools plugin @EXECUTABLE@
 ## Plugin section
 #set $section = $sec_plugin
 --
--t '$section.target_gt'
+--target_gt '$section.target_gt'
```
Comment (nsoranzo) [inline tools/bcftools/bcftools_plugin_setgt.xml:36]:
```suggestion
--target-gt '$section.target_gt'
```

File: tools/bcftools/bcftools_plugin_setgt.xml [hunk 3]
```diff
@@ -52,11 +52,16 @@ bcftools plugin @EXECUTABLE@
...
                 <option value="a">a all genotypes</option>
+                <option value="b">b heterozygous genotypes failing two-tailed binomial test</option>
+                <option value="q">q select genotypes using -i/-e options</option>
+
             </param>
             <param name="new_gt" type="select" label="New genotypes to set">
                 <option value=".">. missing (. or ./., keeps ploidy)</option>
-                <option value="0">0 reference allele</option>
+                <option value="0">0 reference allele (e.g. 0/0 or 0, keeps ploidy)</option>
+                <option value="c:GT">c:GT custom genotype (e.g. 0/0, 0, 0/1, m/M, overrides ploidy)</option>
```
Comment (nsoranzo) [inline tools/bcftools/bcftools_plugin_setgt.xml:62]: I think these new options require bcftools >=1.13 , while currently the wrapper uses 1.10.

Comment (bernt-matthias) [reply tools/bcftools/bcftools_plugin_setgt.xml:62]: There are also a few other open PRs regarding bcftools: https://github.com/galaxyproject/tools-iuc/pulls?q=is%3Apr+is%3Aopen+bcftools

Comment (gallardoalba) [reply tools/bcftools/bcftools_plugin_setgt.xml:62]: Yes thanks, then I think I'll use this PR for updating the tool. Should I include the other PRs also here?

Comment (bernt-matthias) [reply tools/bcftools/bcftools_plugin_setgt.xml:62]: > Should I include the other PRs also here?

That would be awesome.

**Comment by gallardoalba** [comment]
https://github.com/galaxyproject/galaxy/pull/14605 required for including https://github.com/galaxyproject/tools-iuc/pull/4437

File: tools/bcftools/bcftools_plugin_setgt.xml (hunk 3 above)
Comment (gallardoalba) [reply tools/bcftools/bcftools_plugin_setgt.xml:61]: I think it is done @nsoranzo @bernt-matthias.

**Review by bgruening — APPROVED** [review]

**Comment by bgruening** [comment]
@bernt-matthias do you think this can be merged?

File: tools/bcftools/.shed.yml [hunk 4]
```diff
@@ -1,4 +1,5 @@
----
+name: bcftools
```
Comment (bernt-matthias) [inline tools/bcftools/.shed.yml:1]: is name really needed here?

File: tools/bcftools/bcftools_annotate.xml [hunk 5]
```diff
@@ -1,5 +1,5 @@
 <?xml version='1.0' encoding='utf-8'?>
-<tool name="bcftools @EXECUTABLE@" id="bcftools_@EXECUTABLE@" version="@TOOL_VERSION@+galaxy1">
+<tool name="bcftools @EXECUTABLE@" id="bcftools_@EXECUTABLE@" version="@TOOL_VERSION@+galaxy@VERSION_SUFFIX@" profile="20.01">
```
Comment (bernt-matthias) [inline tools/bcftools/bcftools_annotate.xml:2]: making profile another token would have the advantage to allow a profile switch for all tools

File: tools/bcftools/bcftools_annotate.xml [hunk 6]
```diff
@@ -130,6 +137,15 @@ $sec_annofile.annofile.header_lines.__str__.strip()#slurp
...
             <param argument="--mark-sites" type="text" value="" label="Mark Sites TAG"
                 help="add INFO/TAG flag to sites which are (&quot;+&quot;) or are not (&quot;-&quot;) listed in the annotations file" />
+            <param argument="--min-overlap" type="text" value="" label="Minimum required overlap of intersecting regions" optional="true" help="Minimum overlap required as a fraction of the variant in the annotation -a file (ANN), in the target VCF file (:VCF), or both for reciprocal overlap (ANN:VCF). By default overlaps of arbitrary length are sufficient.">
+                <sanitizer invalid_char="">
+                    <valid initial="string.digits">
+                        <add value=":" />
+                        <add value="." />
+                    </valid>
+                </sanitizer>
+                <validator type="regex">[0-9a-zA-Z.:]+</validator>
```
Comment (bernt-matthias) [inline tools/bcftools/bcftools_annotate.xml:147]: Why does the validator contain `a-z` and `A-Z`? 

Wondering if the combination of sanitizer (which removes/replaces all non valid characters) and the validator is useful.

File: tools/bcftools/bcftools_concat.xml [hunk 7]
```diff
@@ -88,8 +91,13 @@ ${section.compact_PS}
                         </when>
                         <when value="no"/>
                     </conditional>
+                    
                     <param name="ligate" type="boolean" truevalue="--ligate" falsevalue="" label="Ligate"
                            help="Ligate phased VCFs by matching phase at overlapping haplotypes" />
+                    <param name="ligate_mode" type="select" optional="true" label="Ligate mode" help="It allows finer control of --ligate behavior">
+                        <option value="--ligate-warn">Ligate warn: Drop sites that are present in one chunk but absent in the other</option>
```
Comment (bernt-matthias) [inline tools/bcftools/bcftools_concat.xml:98]: Maybe add the corresponding command line option ot the end of the options text in parentheses

File: tools/bcftools/bcftools_convert_from_vcf.xml [hunk 8]
```diff
@@ -158,22 +167,22 @@ $convert.sex_info.sex_info_lines.__str__.strip().replace(' ','\t')#slurp
             <param name="input_file" ftype="vcf" value="convert.vcf" />
             <param name="convert_to"  value="gen_sample" />
             <param name="tag"  value="GP" />
-            <param name="chrom"  value="True" />
+            <param name="3N6"  value="True" />
```
Comment (bernt-matthias) [inline tools/bcftools/bcftools_convert_from_vcf.xml:170]: This should be `convert_3N6`

File: tools/bcftools/bcftools_csq.xml [hunk 9]
```diff
@@ -121,7 +126,42 @@ $sec_csq_opts.local_csq
                 </assert_contents>
             </output>
         </test>
-
+        <!-- Test protein seq prediction-->
+        <test>
+            <param name="input_file" ftype="vcf" dbkey="?" value="csq.vcf" />
```
Comment (bernt-matthias) [inline tools/bcftools/bcftools_csq.xml:131]: Wondering why `dbkey` is set to `?`

File: tools/bcftools/bcftools_filter.xml [hunk 10]
```diff
@@ -70,11 +71,22 @@ bcftools @EXECUTABLE@
...
-                ]]></help>
-            </param>
+            <conditional name="conditional_soft_filter">
+                <param name="selector" type="select" label="Perform soft filtering">
+                    <option value="disabled">Disabled</option>
+                    <option value="enabled">Enabled</option>
+                </param>
+                <when value="disabled"/>
+                <when value="enabled">
+                    <param name="soft_filter" type="text" value="" label="Soft Filter" optional="true">
```
Comment (bernt-matthias) [inline tools/bcftools/bcftools_filter.xml:81]: Would be cool if all new parameters would have an `argument` attribute. 

Ideally all parameters would have one.

File: tools/bcftools/bcftools_gtcheck.xml [hunk 11]
```diff
@@ -47,9 +47,11 @@ bcftools @EXECUTABLE@
 
 ${section.all_sites}
 
-#if $section.GTs_only:
-  --GTs-only '${section.GTs_only}'
-#end if
+## The GTs-ony option have been deprecated
```
Comment (bernt-matthias) [inline tools/bcftools/bcftools_gtcheck.xml:50]: Just remove it .. also the commented parameter in the inputs and tests

File: tools/bcftools/bcftools_annotate.xml (hunk 6 above)
Comment (gallardoalba) [reply tools/bcftools/bcftools_annotate.xml:147]: I thought it provides an additional layer of security, but probably you are right. Regarding the alphabetic character, it was a mistake.

File: tools/bcftools/bcftools_annotate.xml (hunk 5 above)
Comment (gallardoalba) [reply tools/bcftools/bcftools_annotate.xml:2]: Sure.

File: tools/bcftools/bcftools_csq.xml (hunk 9 above)
Comment (gallardoalba) [reply tools/bcftools/bcftools_csq.xml:131]: Indeed it is not necessary, not sure why it was included in the previous tests.

**Comment by gallardoalba** [comment]
I think I included all your changes @bernt-matthias, thanks again for the review!

**Comment by bernt-matthias** [comment]
Thanks @gallardoalba

**Pull request merged**